recordingclient.go

```diff
@@ -0,0 +1,53 @@
+package lksdk
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/livekit/protocol/auth"
+	"github.com/twitchtv/twirp"
+
+	livekit "github.com/livekit/server-sdk-go/proto"
+)
+
+type RecordingServiceClient struct {
+	livekit.RecordingService
+	apiKey    string
+	apiSecret string
+}
+
+func NewRecordingServiceClient(url string, apiKey string, secretKey string) *RecordingServiceClient {
+	client := livekit.NewRecordingServiceProtobufClient(url, &http.Client{})
+	return &RecordingServiceClient{
+		RecordingService: client,
+		apiKey:           apiKey,
+		apiSecret:        secretKey,
+	}
+}
+
+func (c *RecordingServiceClient) StartRecording(ctx context.Context, req *livekit.StartRecordingRequest) (*livekit.RecordingResponse, error) {
+	ctx, err := c.withAuth(ctx, auth.VideoGrant{RoomAdmin: true})
+	if err != nil {
+		return nil, err
+	}
+	return c.RecordingService.StartRecording(ctx, req)
+}
+
+func (c *RecordingServiceClient) EndRecording(ctx context.Context, req *livekit.EndRecordingRequest) (*livekit.RecordingResponse, error) {
+	ctx, err := c.withAuth(ctx, auth.VideoGrant{RoomAdmin: true})
+	if err != nil {
+		return nil, err
+	}
+	return c.RecordingService.EndRecording(ctx, req)
+}
+
+func (c *RecordingServiceClient) withAuth(ctx context.Context, grant auth.VideoGrant) (context.Context, error) {
+	at := auth.NewAccessToken(c.apiKey, c.apiSecret)
```

Comment:
would be great to use the same helper vs copying code. can we move this, apikey/secret to an authhelper class that we can composite?